annotations/cut.json

```diff
@@ -1,33 +1,56 @@
 {
-    "command": "cut",
-    "cases":
-    [
-        {
-            "predicate":
-            {
-                "operator": "and",
-                "operands":
-                [
-                    {
-                        "operator": "val_opt_eq",
-                        "operands": ["-d", "--delimiter", "\n"]
-                    },
-                    {
-                        "operator": "exists",
-                        "operands": ["-f", "--fields"]
-                    }
-                ]
-            },
-            "class": "pure",
-            "inputs": ["stdin"],
-            "outputs": ["stdout"],
-			"comments": "Stateless in all cases with exception in case where newline is a delimiter."
-        },
-        {
-            "predicate": "default",
-            "class": "stateless",
-            "inputs": ["stdin"],
-            "outputs": ["stdout"]
-        }
-    ]
+  "command": "cut",
+  "cases": [
+    {
+      "predicate": {
+        "operator": "or",
+        "operands": [
+          {
+            "operator": "val_opt_eq",
+            "operands": [
+              "-d",
+              "\n"
+            ]
+          },
+          {
+            "operator": "exists",
+            "operands": [
+              "-z"
+            ]
+          }
+        ]
+      },
+      "class": "pure",
+      "inputs": [
+        "args[:]"
+      ],
+      "outputs": [
+        "stdout"
+      ]
+    },
+    {
+      "predicate": "default",
+      "class": "stateless",
+      "inputs": [
+        "args[:]"
+      ],
+      "outputs": [
+        "stdout"
+      ]
+    }
+  ],
+  "options": [
+    "stdin-hyphen",
+    "empty-args-stdin"
```

Comment:
Similarly, this is not implemented at the moment (and is very crucial for all the commands that read from stdin if they have no arguments). In its current form, the annotation leads to test failures because uses of `cut` that read from `stdin` are not supported (that is why previously `cut`'s annotation only reads from stdin).

Reply:
This is probably the reason why tests fail.

Reply:
OK there are probably multiple reasons why the tests are failing:P